micmro/performance-bookmarklet
# Page metrics can now be persisted across refreshes

## What? Why?
I couldn't find a tool (bookmarklet or extension) that would persist page performance metrics between page refreshes and ideally, I would like to copy the output into a spreadsheet and create pivot tables to analyze the data pre/post page modification.

This PR creates a ```PageMetric``` class that stores a very small subset of the data collected to localStorage so that the performance between page loads can be analyzed and compared.

This data is exposed via ```PerformanceBookmarklet.PageMetric.dump()```.

Here's what it looks like in action:
![page-metrics-feature](https://cloud.githubusercontent.com/assets/1090773/9078320/dfaeabb6-3b01-11e5-92e0-5bfc0e770998.gif)

What do you think?

## Additional Deploy Steps?
None that I know of. Is there any special testing that I need to do to ensure compatibility with the browser extensions?

## How was it tested?
* I created a local HTML page and injected the performanceBookmarklet.js into the page.
* Run ```PerformanceBookmarklet.PageMetric.dump(",", false);``` from the console. 
  * Verify the page metrics are logged to the console as CSV.
  * Verify the localStorage key ```performance-bookmarklet-metrics``` is still present.
* Run ```PerformanceBookmarklet.PageMetric.dump();``` from the console.
  * Verify the page metrics are logged to the console as TSV.
  * Verify the localStorage key ```performance-bookmarklet-metrics``` is NOT present.
